tests/test_git_repo_scanner.py

```diff
@@ -0,0 +1,225 @@
+# pylint: disable=protected-access
+import unittest
+from unittest import mock
+
+import git
+
+from tartufo import scanner, types
+from tartufo.types import GitOptions
+
+from tests.helpers import generate_options
+
+
+class ScannerTestCase(unittest.TestCase):
+    def setUp(self) -> None:
+        self.options = generate_options(GitOptions)
+
+
+class RepoLoadTests(ScannerTestCase):
+    @mock.patch("git.Repo")
+    def test_repo_is_loaded_on_init(self, mock_repo: mock.MagicMock):
+        scanner.GitRepoScanner(self.options, ".")
+        mock_repo.assert_called_once_with(".")
+
+    @mock.patch("git.Repo")
+    def test_load_repo_loads_new_repo(self, mock_repo: mock.MagicMock):
+        test_scanner = scanner.GitRepoScanner(self.options, ".")
+        test_scanner.load_repo("../tartufo")
+        mock_repo.assert_has_calls((mock.call("."), mock.call("../tartufo")))
+
```

Comment:
Similar to the `issues` conversation above, I think this is something that's going to shake itself out as it gets wired in. It may be slightly over-engineered at the moment, and I will work to fix that part. I was trying to abstract out a little bit to allow for solving many use cases in the future, such as:

1. Load a previously cloned local repo
1. Clone a remote repo into a temporary directory and then load that repo
1. Load a bare repo
1. Scanning an entire GitHub org, which would require cloning and loading each repo individually. Although as I've thought further on this, it's likely best to instantiate a new repo scanner per repo.

I would love more discussion on all of this! I'm only one person, so can only present my own views and thoughts. Those conversations can happen here in tickets, or on the GoDaddy-OSS Slack in #tartufo!

Reply:
👍  Will follow up on this on the #tartufo channel in GoDaddy-OSS Slack.